ui/src/timeMachine/reducers/index.ts

```diff
@@ -981,8 +992,11 @@ const convertView = (
   outType: ViewType
 ): View<QueryViewProperties> => {
   const newView: any = createView(outType)
-
   newView.properties.queries = cloneDeep(view.properties.queries)
+  if (view.properties.type === 'table' && view.properties.fieldOptions) {
```

Comment:
@desa and I would love some feedback on how to run a specific condition based on which type a property is. This was a workaround

Reply:
like a switch?

Reply:
Something like
```ts
interface Square {
    kind: "square";
    size: number;
}
interface Rectangle {
    kind: "rectangle";
    width: number;
    height: number;
}
interface Circle {
    kind: "circle";
    radius: number;
}

type Shape = Square | Rectangle | Circle;

...

    switch (s.kind) {
        case "square": return s.size * s.size;
        case "rectangle": return s.height * s.width;
        case "circle": return Math.PI * s.radius ** 2;
    }
```

where were I don't have to switch on `kind` and can switch on the interface instead like
```ts
    switch (s.kind) {
        case Square: return s.size * s.size;
        case Rectangle: return s.height * s.width;
        case Circle: return Math.PI * s.radius ** 2;
    }
```

Reply:
Here's an example from go https://tour.golang.org/methods/16